### PR TITLE
Feat/multiple brains files

### PR DESCRIPTION
**Pull request by gozineb**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context.

## Checklist before requesting a review

Please delete options that are not relevant.

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented hard-to-understand areas
- [ ] I have ideally added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged

### CODE REVIEW

File: backend/llm/PrivateBrainPicking.py [hunk 1]
```diff
@@ -32,23 +16,24 @@ class PrivateBrainPicking(BrainPicking):
...
+        :param brain_id: The user id to be used for CustomSupabaseVectorStore.
         :return: PrivateBrainPicking instance
         """
         # Call the parent class's initializer
         super().__init__(
             model=model,
-            user_id=user_id,
+            brain_id=brain_id,
             chat_id=chat_id,
+            brain_i=brain_id,
```
Comment (mamadoudicko) [inline backend/llm/PrivateBrainPicking.py:36]: ❌

File: backend/models/brains.py [hunk 2]
```diff
@@ -24,6 +28,26 @@ def commons(self) -> CommonsDep:
...
+
+        print('current_brain_size', current_brain_size)
+        return current_brain_size
+
+    # To keep ??
+    @property
+    def brain_filling(self):
+        if self.max_brain_size == 0:
+            return 0
+        return self.brain_size / self.max_brain_size
```
Comment (mamadoudicko) [inline backend/models/brains.py:44]: Not used for now

File: backend/routes/brain_routes.py [hunk 3]
```diff
@@ -78,7 +90,7 @@ async def delete_brain_endpoint(brain_id: UUID):
     """
     Delete a specific brain by brain ID.
     """
-    brain = Brain(brain_id=brain_id)
+    brain = Brain(id=brain_id)
```
Comment (StanGirard) [inline backend/routes/brain_routes.py:93]: Make sure it is the users brain

File: backend/routes/crawl_routes.py [hunk 4]
```diff
@@ -1,47 +1,37 @@
...
     """
     Crawl a website and process the crawled data.
     """
-    commons = common_dependencies()
-    max_brain_size = os.getenv("MAX_BRAIN_SIZE")
-    if request.headers.get('Openai-Api-Key'):
-        max_brain_size = os.getenv("MAX_BRAIN_SIZE_WITH_KEY",209715200)
 
-    user_unique_vectors = get_unique_user_data(commons, current_user)
+    # //Check the rights user_id and
```
Comment (StanGirard) [inline backend/routes/crawl_routes.py:25]: and ?

File: backend/routes/explore_routes.py [hunk 5]
```diff
@@ -70,7 +53,7 @@ async def download_endpoint(
             "metadata->>file_name, metadata->>file_size, metadata->>file_extension, metadata->>file_url",
             "content",
         )
-        .match({"metadata->>file_name": file_name, "user_id": current_user.email})
+        .match({"metadata->>file_name": file_name})
```
Comment (StanGirard) [inline backend/routes/explore_routes.py:56]: If two files same name it will match

File: backend/utils/brains.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+from fastapi import APIRouter
+from logger import get_logger
+from models.brains import Brain, BrainToUpdate
+from models.settings import CommonsDep
+from models.users import User
+
+logger = get_logger(__name__)
+
+brain_router = APIRouter()
+
```
Comment (StanGirard) [inline backend/utils/brains.py:12]: turn to class later on

File: backend/vectorstore/supabase.py [hunk 7]
```diff
@@ -3,23 +3,24 @@
...
 from langchain.vectorstores import SupabaseVectorStore
+
 from supabase import Client
 
 
 class CustomSupabaseVectorStore(SupabaseVectorStore):
     """A custom vector store that uses the match_vectors table instead of the vectors table."""
 
-    user_id: str
+    brain_id: str = "none"
```
Comment (StanGirard) [inline backend/vectorstore/supabase.py:13]: Why at none by default ?

File: scripts/20230620183620_use_supabase_user_id.sql [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+-- WHERE 
+--     conrelid::regclass = 'users'::regclass;
+
+ALTER TABLE users RENAME COLUMN user_id TO old_user_id;
+ALTER TABLE users RENAME COLUMN supabase_id TO user_id;
+
+--- Create a new primary key with user_id and date
+ALTER TABLE users ADD PRIMARY KEY (user_id, date);
+
+-- Delete duplicate rows from the 'users' table
```
Comment (StanGirard) [inline scripts/20230620183620_use_supabase_user_id.sql:34]: Make it idempotent

File: scripts/20230627151100_update_match_vectors.sql [hunk 9]
```diff
@@ -0,0 +1,38 @@
+-- Migration script
+BEGIN;
+
+-- Drop the old function if exists
```
Comment (StanGirard) [inline scripts/20230627151100_update_match_vectors.sql:4]: Make it idempotent

File: scripts/20230627151100_update_match_vectors.sql [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+DROP FUNCTION IF EXISTS match_vectors(VECTOR(1536), INT, TEXT);
+
+-- Create the new function
+CREATE OR REPLACE FUNCTION match_vectors(query_embedding VECTOR(1536), match_count INT, p_brain_id UUID)
```
Comment (StanGirard) [inline scripts/20230627151100_update_match_vectors.sql:8]: Merge all migrations into one
Ask GPT ;)

File: frontend/lib/components/NavBar/components/NavItems/index.tsx [hunk 11]
```diff
@@ -24,8 +24,8 @@ export const NavItems = ({
 }: NavItemsProps): JSX.Element => {
   const { session } = useSupabase();
   const isUserLoggedIn = session?.user !== undefined;
-  const shouldUseMultipleBrains = useFeature('multiple-brains').on;
-  
+  const shouldUseMultipleBrains = useFeature("multiple-brains").on;
```
Comment (mamadoudicko) [inline frontend/lib/components/NavBar/components/NavItems/index.tsx:27]: We should may be now remove this ?

**Review by mamadoudicko — APPROVED** [review]
Very nice job ! 

Will soon be able to share my brain 🤏🏽 with others 😍

Let's rerun the migration in a dev environment one last time.

**Pull request merged**